### ISSUE TITLE
Stop SPAM!

### ISSUE THREAD

**Issue by batumibiz** [opencart/opencart#12616]
I appeal to @TheCartpenter
Why do you create a lot of PR for a trivial reason, like removing an extra space, etc.?
Can't you do it all in 1 PR?
Because of your PR attack, notifications from GitHub ended up in spam.

**Comment by stalker780** [comment]
He is an idiot. He does not understand what he is doing.
This lasts for years. And there is a number of complaints from opencart subscribers.

**Comment by batumibiz** [comment]
It seems to me that the most useful thing is to block it, it is impossible to endure it any longer.

**Comment by TheCartpenter** [comment]
Since the subject has been mentioned before, and that none of you bothered to take a look at my commits list situation, I will submit it again: https://www.github.com/TheCartpenter/opencart/commits

As none of you noticed, and as you can see now, I am stuck with red Xes. I tried the proposed solutions on this repository in the past and none of it worked. Which is why, my only option is to post individual files for each commit since I cannot push the changes from my repository even though the codes need to be compared prior to create a pull request nor is it against Github regulations to create separate commits.

If notifications annoys every each of you, then by all means, since this is my only option, feel free to ignore it but the fact that you decide to ignore those notifications will make it less clear to all of you not only why this keeps happening but also ignoring other notifications.



> I appeal to @TheCartpenter Why do you create a lot of PR for a trivial reason, like removing an extra space, etc.? Can't you do it all in 1 PR? Because of your PR attack, notifications from GitHub ended up in spam.

No, unfortunately I can't unless a definite solution to resolve those red Xes comes to light. Again, it is NOT a PR attack since it is not against regulations to post individual commits on Github. What you believe is simply a matter of opinion.

**Comment by mhcwebdesign** [comment]
Several users have pointed out to you in the past what git clients you can use, to stage changed files to commit, and then to trigger a commit on one go.

I am just ignoring your daily high volume pull requests.

**Comment by TheCartpenter** [comment]
> Several users have pointed out to you in the past what git clients you can use, to stage changed files to commit, and then to trigger a commit on one go.
> 
> I am just ignoring your daily high volume pull requests.

And those solutions have been followed and tried, the results are still showing red Xes on the repository, since. Therefore, it is up to your total discretion to ignore it.

**Comment by stalker780** [comment]
He is sure he is fine with such PRs in the future. 
That's not his problem. 
Instead of sorting out what is wrong with his software, PRs or contacting GitHub support he will continue spamming OC subscribers.

**Comment by TheCartpenter** [comment]
It's not with the software.

**Comment by batumibiz** [comment]
@TheCartpenter The fact that you can't figure out your software is your problem.
But you create serious inconvenience for other users.
I understand that you are going to continue to spam in PR?

**Comment by TheCartpenter** [comment]
Alright folks, it's working now: https://github.com/opencart/opencart/pull/12619 .

**Comment by batumibiz** [comment]
> Should be good on the commit's list now. The tree is now in-sync with the MB. Has nothing to do with the software. The issue was from the website's account directly!
> 
> Github integrated new features and I am sure we all need to be aware of them so this doesn't happen again!
> 
> I'll try again with the multi-files posting in one commit to see if it works now.

There were no problems with GitHub before.

**Comment by batumibiz** [comment]
What you do (code formatting) can be done automatically in a few minutes.
In PhpStorm, we set up formatting according to PSR-12, process all the code and rejoice.

I have now checked that ALL OpenCart source code is formatted under the PSR-12 standard:
- it took 2 minutes.
- formatted 857 files

Do you think that I can now create 857 PR?
So I will probably immediately go to the leading position in terms of activity and get all the achievements! :)

**Comment by batumibiz** [comment]
> Alright folks, it's working now: #12619 .

Are you editing the source code through the GitHub Web interface?

![1526059477246_0](https://github.com/opencart/opencart/assets/10770779/073786cb-1696-4705-b868-91feb0ee0b83)

**Comment by TheCartpenter** [comment]
> > Alright folks, it's working now: #12619 .
> 
> Are you editing the source code through the GitHub Web interface?
> 
> ![1526059477246_0](https://user-images.githubusercontent.com/10770779/260012138-073786cb-1696-4705-b868-91feb0ee0b83.gif)

No, which is why I stated that it's not a software issue. It was directly linked from the Github Profile!

**Comment by TheCartpenter** [comment]
> What you do (code formatting) can be done automatically in a few minutes. In PhpStorm, we set up formatting according to PSR-12, process all the code and rejoice.
> 
> I have now checked that ALL OpenCart source code is formatted under the PSR-12 standard:
> 
> * it took 2 minutes.
> * formatted 857 files
> 
> Do you think that I can now create 857 PR? So I will probably immediately go to the leading position in terms of activity and get all the achievements! :)

It wasn't it. The restriction was implied directly from the Github Profile itself!

**Comment by TheCartpenter** [comment]
It's all-in-one pull request now.

**Comment by batumibiz** [comment]
> It's all-in-one pull request now.

Another question: why don't you merge everything into one commit?
For the sake of adding one space, you make a separate commit.
Are you trying to cheat statistics?

**Comment by TheCartpenter** [comment]
> > It's all-in-one pull request now.
> 
> Another question: why don't you merge everything into one commit? For the sake of adding one space, you make a separate commit. Are you trying to cheat statistics?
> 
> At this pace, he will overtake Daniel before the end of the year according to the statistics of committees.

No cheating intended as the actual reasons were explained on the above with the red Xes until I could solve the issue. Beside, as also said on the above, the commits have indeed been merged accordingly, since.

**Comment by TheCartpenter** [comment]
> > Beside, as also said on the above, the commits have indeed been merged accordingly, since.
> 
> You are deceiving us. There are 67 (!!!) not significant commits in your PR Although it was possible to put all these little things in 1 commit.

If this is what you believe, there isn't much I can do about that but one thing I can confirm is that I was not deceiving anybody for that to happen. Quite frankly, I could care less about the stats. I just want to solve these picks as soon as possible in order for all of us to move on with these codes. If you believe that these picks have not been posted for the right purpose, then why didn't anybody else addressed those issues before whereas other controllers, languages, views and even libraries files are indeed well-coded from the rest of those locations? !

At least, I took the time to post these 67 commits and merged them altogether since the issue was resolved today due to a rule policy that was restricting me to fork for an unknown reason. Since then, I am able to merge. As explained on the merged pull request I created, requested by all of you, unless you want me to fall back to the way it was, then there's no reason for those critics to be posted about since I did solved the issue since.

**Comment by mhcwebdesign** [comment]
Thank you @TheCartpenter for looking into it and solving it.

I am now closing this issue.

**Comment by TheCartpenter** [comment]
Please is all mine!

**Comment by batumibiz** [comment]
@danielkerr @mhcwebdesign the spam attack from TheCartpenter to PR has begun again.
i propose to block this user, or create a vote on this matter.
As I see it, he **does not even understand that he is wrong** and spam from him will constantly go.
The harm and inconvenience from it is much greater than the benefits.

**Comment by danielkerr** [comment]
We are not. I will go through them

**Comment by danielkerr** [comment]
Takes me 10mins

**Comment by mhcwebdesign** [comment]
@TheCartpenter I genuinely thought you had solved the issue and were able to combine them all to a single pull request. So what went wrong? Please use a proper git client!

**Comment by danielkerr** [comment]
I prefer he leave it simple so I can see what's being changed

**Comment by TheCartpenter** [comment]
> @TheCartpenter I genuinely thought you had solved the issue and were able to combine them all to a single pull request. So what went wrong? Please use a proper git client!

As you seem to be pretty good at repeating the same questions, please read my replies on the above. Besides, as per Daniel, he just mentioned he prefers to leave it simple. Therefore, and from now on, this is the path that will be taken.

**Comment by batumibiz** [comment]
> I prefer he leave it simple so I can see what's being changed

He sent over 60 PRs.
Of these, 50 or more are formatting-only, not affecting code execution in any way. In one file he added a space, in another he removed it. And for each action, he creates a separate commit and sends a separate PR.
All formatting could be combined into 1 commit and 1 PR.

If you need code formatting (for example PSR-2), this is done immediately and automatically.
Why create a commit and PR for the sake of one space?

**Comment by mhcwebdesign** [comment]
@batumibiz : Thank goodness we don't have OCmod/VQmod based modifications anymore on these files, it would have broken the modifications in mass !

**Comment by TheCartpenter** [comment]
> @batumibiz : Thank goodness we don't have OCmod/VQmod based modifications anymore on these files, it would have broken the modifications in mass !

And the reason why being that doing so doesn't interfere with other projects since OC v4,x releases, still on this day.

**Comment by batumibiz** [comment]
@danielkerr @mhcwebdesign I resented spam because if you accept all these garbage commits, the GIT history will be heavily littered with meaningless commits. This makes it very difficult to deal with really serious changes.

It is generally useful to prohibit such commits (where a space is added / removed).
Or, officially declare that OpenCart adheres to the PSR-2 coding standard (for example), reformat all the code once (this is done easily and automatically) and then, in accepted PRs, require compliance with the code format.
In this case, it will be useful to use squizlabs/PHP_CodeSniffer in conjunction with GitHub Actions CI. Then all PRs will be automatically checked for compliance with the coding standard.

**Comment by TheCartpenter** [comment]
> @danielkerr @mhcwebdesign I resented spam because if you accept all these garbage commits, the GIT history will be heavily littered with meaningless commits. This makes it very difficult to deal with really serious changes.
> 
> It is generally useful to prohibit such commits (where a space is added / removed). Or, officially declare that OpenCart adheres to the PSR-2 coding standard (for example), reformat all the code once (this is done easily and automatically) and then, in accepted PRs, require compliance with the code format. In this case, it will be useful to use squizlabs/PHP_CodeSniffer in conjunction with GitHub Actions CI. Then all PRs will be automatically checked for compliance with the coding standard.

Then, those commits aren't as meaningless as one would say since it is also suggested underneath to other compliance tools in compliance to coding standards altogether.

**Comment by nexadomain** [comment]
@TheCartpenter
Why don't you try **Github Desktop** App?
